File: website/src/app/localisation.service.ts
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class LocalisationService {

  static translations: { [id: string]: string } | undefined;

  getText(label: string, defaultValue?: string): string {
    if (!label || label == "@LOC_PLACEHOLDER" || label == "@LOC_UNINITIALIZED" || label == "@LOC_EMPTY") return defaultValue || "";

    if (label.startsWith("@")) label = label.substring(1);

    let text = LocalisationService.translations ? LocalisationService.translations[label] : undefined;
    if (!text) return defaultValue || label;

    return text.replace(/\\n/g, "\n");
  }

  static SetLabels(translations: { [id: string]: string }) {
    this.translations = translations;
  }
}

File: Loader/LocalisationService.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Loader
{
	public class LocalisationService
	{
		Dictionary<string, string> labels;

		public LocalisationService(Dictionary<string, string> labels)
		{
			this.labels = labels;
		}

		public string GetText(string label, string fallback = null)
		{
			if (label == null) return fallback;

			var key = label.StartsWith("@") ? label.Substring(1) : label;

			if (key == "LOC_EMPTY") return fallback;
			if (key == "LOC_UNINITIALIZED") return fallback;

			if (!labels.ContainsKey(key)) return fallback;

			var text = labels[key].Replace("\\n", "\n");

			if (text == "<= PLACEHOLDER =>") return fallback ?? label;
			if (String.IsNullOrWhiteSpace(text)) return fallback ?? label;

			return text;
		}
	}
}
